src/unitxt/operators.py

```diff
@@ -508,75 +508,124 @@ def process_value(self, value: Any) -> Any:
+    prefixes: Optional[Union[List[str], Dict[str, int]]] = {
+        " ": 20,
+        "\\t": 10,
+        "\\n": 40,
+        "": 30,
+    }
+    prefix_len: Optional[int] = 3
+    suffixes: Optional[Union[List[str], Dict[str, int]]] = {
+        " ": 20,
+        "\\t": 10,
+        "\\n": 40,
+        "": 30,
+    }
+    suffix_len: Optional[int] = 3
+    remove_existing_whitespaces: Optional[bool] = False
 
     def verify(self):
         assert (
-            isinstance(self.suffixes, list) or isinstance(self.suffixes, dict)
-        ), f"Argument 'suffixes' should be either a list or a dictionary, whereas it is of type {type(self.suffixes)}"
-
-        if isinstance(self.suffixes, dict):
-            for k, v in self.suffixes.items():
-                assert isinstance(
-                    k, str
-                ), f"suffixes should map strings, whereas key {k!s} is of type {type(k)}"
-                assert isinstance(
-                    v, int
-                ), f"suffixes should map to ints, whereas value {v!s} is of type {type(v)}"
-        else:
-            for k in self.suffixes:
-                assert isinstance(
-                    k, str
-                ), f"suffixes should be a list of strings, whereas member {k!s} is of type {type(k)}"
+            self.prefixes or self.suffixes
+        ), "At least one of prefixes/suffixes should be not None."
+        for arg, arg_name in zip(
+            [self.prefixes, self.suffixes], ["prefixes", "suffixes"]
+        ):
+            assert (
+                arg is None or isinstance(arg, list) or isinstance(arg, dict)
+            ), f"Argument {arg_name} should be either None or a list or a dictionary, whereas it is of type {type(arg)}"
+
+            if arg is None:
+                continue
+            if isinstance(arg, dict):
+                for k, v in arg.items():
+                    assert isinstance(
+                        k, str
+                    ), f"{arg_name} should map strings, whereas key {k!s} is of type {type(k)}"
+                    assert isinstance(
+                        v, int
+                    ), f"{arg_name} should map to ints, whereas value {v!s} is of type {type(v)}"
+            else:
+                for k in arg:
+                    assert isinstance(
+                        k, str
+                    ), f"{arg_name} should be a list of strings, whereas member {k!s} is of type {type(k)}"
+        assert (
+            self.prefix_len > 0
+        ), f"prefix_len must be positive, got {self.prefix_len}"
+        assert (
+            self.suffix_len > 0
+        ), f"suffix_len must be positive, got {self.suffix_len}"
         super().verify()
 
-    def prepare(self):
-        self.pats = (
-            self.suffixes
-            if isinstance(self.suffixes, list)
-            else [k for k, v in self.suffixes.items()]
+    def _calculate_distribution(self, end, end_name):
```

Comment:
Hi Dafna, I think this kind of implicit assigned through names is harder to read.  Better to have functions that receive input  and not modify them and then return outputs.  Then the caller can assign the value to the right field.

Reply:
Done. Now I finally got it what you said in the first place..